src/ui/App.swift

```diff
@@ -248,7 +248,7 @@ class App: AppCenterApplication, NSApplicationDelegate {
             isFirstSummon = false
             if Windows.list.count == 0 || CGWindow.isMissionControlActive() { hideUi(); return }
             // TODO: can the CGS call inside detectTabbedWindows introduce latency when WindowServer is busy?
-            Windows.detectTabbedWindows()
+            //Windows.detectTabbedWindows()
```

Comment:
You replaced this unconditional call, before showing the UI, with a similar call, but only when a window gets deleted. I think it will miss many cases where tabs are reconfigured. I was originally tracking tabs in the events (window resized, created, deleted, etc). Then I faced many subtle issues where tabs wouldn't properly be noticed.

I ended up putting an unconditional call upfront before showing the UI, and it proved way more reliable. I'm worried here that you are going back to the old approach, and it will not pick up the right state. The user can combine tabs with the mouse, with keyboard shortcuts, by closing windows, by creating windows, etc.

Reply:
The issue with this new method compared to the old one is that it only works for the current space. So calling it when the UI is displayed would not be sufficient. It is called for the application when a window of the application is focused or deleted. The only case that is not covered that I can think of is when AltTab closes a window that is part of a group of tabs in another space.

Reply:
I'll give you an example:

* Create 2 Finder windows
* Click in the menubar at the top: `Window > Merge All Windows`
* This will not trigger any accessibility events (except `kAXFocusedUIElementChangedNotification` which we used to listen to, but no longer do)
* It will thus never trigger `detectTabsInCurrentSpace()`
* AltTab will not be aware of this change in tabs/windows